connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java

```diff
@@ -107,6 +110,14 @@ public Object answer() throws Throwable {
         assertEquals(200, response.getStatus());
 
         assertEquals(expectedHeader, response.getHeaderString("Access-Control-Allow-Origin"));
+
+        response = request("/connector-plugins/FileStreamSource/validate")
+            .header("Referer", origin + "/page")
+            .header("Origin", origin)
+            .header("Access-Control-Request-Method", method)
+            .options();
+        assertEquals(404, response.getStatus());
```

Comment:
The 404 seemed odd to me, so I want to make sure I understand why we're seeing it (since I would normally interpret that as an error). It seems that CORS doesn't require that you return a particular HTTP status code, so for the preflighted request it's actually ok if we return any code here. In the case of Jetty, it looks like the `CrossOriginFilter`'s `chainPreflight` option defaults to true such that the `OPTIONS` request continues on for normal handling. Since we don't implement `OPTIONS` for any of our endpoints (and the one under test, specifically), we end up with a 404. Is that correct?